@here/harp-mapview/lib/MapView.ts

```diff
@@ -1802,8 +1871,61 @@ export class MapView extends THREE.EventDispatcher {
         // rendering multiple times during a single frame.
         if (this.m_animationFrameHandle !== undefined) {
             cancelAnimationFrame(this.m_animationFrameHandle);
+            this.m_animationFrameHandle = undefined;
         }
-        this.m_animationFrameHandle = requestAnimationFrame(this.handleRequestAnimationFrame);
+
+        if (this.m_maxFps <= 0) {
+            // Render at maximum FPS.
+            this.m_animationFrameHandle = requestAnimationFrame(this.handleRequestAnimationFrame);
+            return;
+        }
+
+        const vSyncFrameTime = 1000 / this.m_detectedFps;
+        const frameInterval = 1000 / this.m_maxFps;
+
+        const startTime =
+            this.m_previousFrameTimeStamp === undefined
+                ? performance.now()
+                : this.m_previousFrameTimeStamp;
+
+        // Magic ingredient to compensate time flux.
+        const fudgeTimeInMs = 3;
+
+        const previousFrameTime =
+            this.m_previousFrameTimeStamp === undefined ? 0 : this.m_previousFrameTimeStamp;
+
+        // Compute a practical value to compare against.
+        const targetTime = previousFrameTime + frameInterval - vSyncFrameTime - fudgeTimeInMs;
+
+        if (startTime > targetTime) {
+            this.detectCurrentFps(startTime);
+
+            this.m_animationFrameHandle = requestAnimationFrame(this.handleRequestAnimationFrame);
+        } else {
+            this.m_previousRequestAnimationTime = startTime;
+            this.m_targetRequestAnimationTime = targetTime;
+
+            this.m_animationFrameHandle = requestAnimationFrame(this.handlePostponedAnimationFrame);
+        }
+    }
+
+    private postponedAnimationFrame(now: number) {
+        if (this.m_targetRequestAnimationTime === undefined) {
+            return;
+        }
+
+        if (this.m_animationFrameHandle !== undefined) {
+            cancelAnimationFrame(this.m_animationFrameHandle);
+            this.m_animationFrameHandle = undefined;
+        }
+
+        this.detectCurrentFps(now);
```

Comment:
is this correct? This will detect the fps using a postponed animation frame, which will artificially inflate the FPS

Reply:
in this case we need detected fps only to measure delay to next rAF callback,
and this delay is computed by browser using previous callbacks duration.
As there is no standard algo for this delay, and no api method, we need to estimate it somehow.

Also, we can't do rAF, then setTimeout (or similar) to next rAF - that won't work as expected, especially on high cpu load.